### PR TITLE
[NIFI-2926] add policy action item to the user table and add user cen…

### PR DESCRIPTION
**Pull request by scottyaslan**
Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [ ] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [ ] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [ ] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

…tric policy dialog with user policy listing table compolete with GoTo action.

### CODE REVIEW

**Comment by mcgilman** [comment]
Reviewing...

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/users/nf-users-table.js [hunk 1]
```diff
@@ -470,6 +470,214 @@ nf.UsersTable = (function () {
...
+            return markup;
+        };
+
+        var userPoliciesColumns = [
+            {id: 'policy', name: 'Policy', sortable: true, resizable: true, formatter: policyDisplayNameFormatter, width: 150},
+            {id: 'action', name: 'Action', sortable: true, resizable: false, formatter: actionFormatter, width: 50}
+        ];
+
+        // add the actions if we're in the shell
+        if ((top !== window) && nf.Common.canAccessPolicies()) {
```
Comment (mcgilman) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/users/nf-users-table.js]: Policies will have their own specific permissions. No need to check if the user can access all policies here.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/users/nf-users-table.js [hunk 2]
```diff
@@ -470,6 +470,214 @@ nf.UsersTable = (function () {
...
+            buttons: [{
+                buttonText: 'Close',
+                color: {
+                    base: '#728E9B',
+                    hover: '#004849',
+                    text: '#ffffff'
+                },
+                handler: {
+                    click: function () {
+                        //TODO: some work
```
Comment (mcgilman) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/users/nf-users-table.js]: Any more work needed here? ;)

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/users/nf-users-table.js [hunk 3]
```diff
@@ -524,6 +732,10 @@ nf.UsersTable = (function () {
                 markup += '<div title="Remove" class="pointer delete-user fa fa-trash"></div>';
             }
 
+            if (nf.Common.isDefinedAndNotNull(dataContext.component.accessPolicies)) {
```
Comment (mcgilman) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/users/nf-users-table.js]: Should probably show the key when the accessPolicies are not empty. I think its just `!nf.Common.isEmpty(dataContext.component.accessPolicies)`.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/users/nf-users-table.js [hunk 4]
```diff
@@ -470,6 +470,214 @@ nf.UsersTable = (function () {
...
+        } else if (resource.startsWith('/remote-process-groups')) {
+            policyLabel += 'remote process group ';
+        } else if (resource.startsWith('/reporting-tasks')) {
+            policyLabel += 'reporting task ';
+        } else if (resource.startsWith('/templates')) {
+            policyLabel += 'template ';
+        }
+
+        if (dataContext.component.componentReference.permissions.canRead === true) {
+            policyLabel += dataContext.component.componentReference.component.name;
```
Comment (mcgilman) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/users/nf-users-table.js]: Thoughts on emphasizing the component name? In some example it seems to blend in with the rest of the string.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/WEB-INF/partials/users/user-policies-dialog.jsp [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+--%>
+<%@ page contentType="text/html" pageEncoding="UTF-8" session="false" %>
+<div id="user-policies-dialog" class="hidden large-dialog">
+    <div class="dialog-content">
+        <div id="user-policy-message">
+            NOTE: Due to the nature of user policies (specifically inheritance) only top level policies are available.
```
Comment (mcgilman) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/WEB-INF/partials/users/user-policies-dialog.jsp]: Thoughts on updating this note to read:

> Listing includes all policies for the selected user. Some policies may be inherited by descendant components unless explicitly overridden.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/nf-common.js [hunk 6]
```diff
@@ -657,7 +695,7 @@ nf.Common = (function () {
                     return nf.Common.escapeHtml(value);
                 }
             } else {
-                return '<span class="unset" style="font-size: 13px; padding-top: 2px;">No value previously set</span>';
+                return '<span class="unset" style="font-size: 13px; padding-top: 2px;">No value set</span>';
```
Comment (mcgilman) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/nf-common.js]: Can we also update the `Empty string previously set` above?

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/nf-common.js [hunk 7]
```diff
@@ -1052,7 +1090,7 @@ nf.Common = (function () {
             if (!nf.Common.isDefinedAndNotNull(rawDateTime)) {
                 return new Date();
             }
-            if (rawDateTime === 'No value previously set') {
+            if (rawDateTime === 'No value set') {
                 return new Date();
             }
             if (rawDateTime === 'Empty string previously set') {
```
Comment (mcgilman) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/nf-common.js]: Can we update the `Empty string previously` set here too?

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/WEB-INF/partials/users/user-policies-dialog.jsp (hunk 5 above)
Comment (scottyaslan) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/WEB-INF/partials/users/user-policies-dialog.jsp]: I like it! Do you think we should leave the "NOTE:" in front of it?

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/users/nf-users-table.js (hunk 4 above)
Comment (scottyaslan) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/users/nf-users-table.js]: Maybe bold or italicized?

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/users/nf-users-table.js (hunk 2 above)
Comment (scottyaslan) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/users/nf-users-table.js]: haha...nope

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/users/nf-users-table.js (hunk 3 above)
Comment (scottyaslan) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/users/nf-users-table.js]: ah....Right bc the accessPolicies are an array...

**Comment by markap14** [comment]
@mcgilman I'd be happy to review this if you need another reviewer given that you and @scottyaslan have both provided commits for this PR.

**Comment by mcgilman** [comment]
@markap14 Sounds great. Thanks!

**Comment by markap14** [comment]
@mcgilman @scottyaslan I've reviewed this and tested that the functionality works as expected. +1 merged to master. Thanks for knocking this out - will definitely be a useful feature.

**Pull request closed without merging**